pennylane/transforms/diagonalize_measurements.py

```diff
@@ -60,6 +60,14 @@ def diagonalize_measurements(tape, supported_base_obs=_default_supported_obs, to
         circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>` transform
         can be applied.
 
+    .. note::
+        This transform will diagonalize what it can, i.e., ``qml.X``, ``qml.Y``, ``qml.Z``,
+        ``qml.Hadamard``, ``qml.Identity``, or a linear combination of them. Any unrecognized
+        observable will not raise an error, deferring to the device's validation for supported
```

Comment:
> Any unrecognized observable will not raise an error, deferring to the device's validation for supported measurements later on.

Should we add an example of this behaviour under Usage Details, showing a circuit where some things are diagonalized, and others are left unchanged?